mediapipe/graphs/instantmotiontracking/calculators/matrices_manager_calculator.cc

```diff
@@ -114,13 +118,32 @@ class MatricesManagerCalculator : public CalculatorBase {
     // This returns a scale factor by which to alter the projection matrix for
     // the specified render id in order to ensure all objects render at a similar
     // size in the view screen upon initial placement
-    const float GetDefaultRenderScale(const int render_id) {
+    const DiagonalMatrix3f GetDefaultRenderScaleDiagonal(const int render_id, const float user_scale_factor, const float gif_aspect_ratio) {
+      float combined_scale = 1.0f;
+      float x_scalar = 1.0f;
+      float y_scalar = 1.0f;
+
       if (render_id == 0) { // GIF
-        return 160.0f;
+        // 160 is the scaling preset to make the GIF asset appear relatively
+        // similar in size to all other assets
+        combined_scale = 160.0f * user_scale_factor;
+        // GIF if wider horizontally (scale on x-axis)
+        if (gif_aspect_ratio >= 1.0f) {
+          x_scalar = gif_aspect_ratio;
```

Comment:
Do you need this log or can you always do "x_scalar = gif_aspect_ratio"?

Reply:
If the aspect ratio is vertical, then the x_scaler value needs to remain 1.0f to make sure we scale vertically and not horizontally. I think the conditional is necessary.